packages/netsuite-adapter/test/change_validators/unique_fields.test.ts

```diff
@@ -0,0 +1,222 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { buildElementsSourceFromElements } from '@salto-io/adapter-utils'
+import { ElemID, InstanceElement, ObjectType, toChange, ReadOnlyElementsSource, Field, Element, ChangeError } from '@salto-io/adapter-api'
+import uniqueFields from '../../src/change_validators/unique_fields'
+import { CUSTOM_RECORD_TYPE, CUSTOM_RECORD_TYPE_PREFIX, FINANCIAL_LAYOUT, METADATA_TYPE, NETSUITE, SAVED_SEARCH } from '../../src/constants'
+
+const DUPLICATED_FIELD = 'test uniqueness'
+const UNIQUE_FIELD = 'test uniqueness 2'
+
+describe('unique fields validator', () => {
+  type ValidTypes = InstanceElement | ObjectType | Field
+
+  type TestElements = {
+    BASIC: ValidTypes
+    SAME_FIELD: ValidTypes
+    DIFF_FIELD: ValidTypes
+  }
+
+  const getTestElements = (typeName: string, createElem:
+    (name: string, elemID: ObjectType, uniqueField: string) => ValidTypes): TestElements => {
```

Comment:
the common style is that if the args don't fit into one line then they should be in separate lines for each arg:
```suggestion
  const getTestElements = (
    typeName: string,
    createElem: (name: string, elemID: ObjectType, uniqueField: string) => ValidTypes
  ): TestElements => {
```

Reply:
and I think that the second arg should be called `type`?